pkg/services/appinsights/types.go

```diff
@@ -0,0 +1,21 @@
+package appinsights
+
+import "github.com/Azure/open-service-broker-azure/pkg/service"
+
+type instanceDetails struct {
+	ARMDeploymentName  string               `json:"armDeployment"`
+	AppInsightsName    string               `json:"appInsightsName"`
+	InstrumentationKey service.SecureString `json:"instrumentationKey"`
```

Comment:
The type of `InstrumentationKey` cannot be `SecureString`. When marshaling to JSON, type `SecureString` will be encrypted, that is, the user will get a encrypted key.

Reply:
In binding, it will be converted to a string. What I not sure is that, is it secret? We don't need use SecureString if not.

Reply:
1. I don't think it is a secret, the value of `InstrumentationKey` is easily found at the main page of an app insight instance in Azure Portal. If it is a secret, it shouldn't be put at the main page.
2. The only difference between `string` and `SecureString` is when converting to JSON, `SecureString` will be encrypted and `string` will not. In memory, they're both bare non-encrypted string. So if `SecureString` is converted to `string` when binding, maybe use `string` at beginning is a better practice.

Reply:
Consulted @HDYA, `InstrumentationKey` is like ID+KEY to the app insights. That makes sense -- anyone can use the app insights instance if he has this key. So, we should use `SecureString` for this field, right?

Reply:
It makes sense. Thanks for the explanation.